ui/src/pages/share/index.vue

```diff
@@ -22,33 +24,40 @@ const handleConfirm = () => {
     return;
   }
 
-  shareCode.value = verifyValue.value;
-  showModal.value = false;
+  connectionStore.setConnectionState({
+    shareCode: verifyValue.value,
+  });
+
+  nextTick(() => {
+    showModal.value = false;
+  });
 };
 </script>
 
 <template>
-  <div>
-    <n-modal
-      v-model:show="showModal"
-      preset="dialog"
-      :title="t('VerifyCode')"
-      :positive-text="t('ConfirmBtn')"
-      :closable="false"
-      :close-on-esc="false"
-      :mask-closable="false"
-      @positive-click="handleConfirm"
-    >
-      <n-input
-        v-model:value="verifyValue"
-        clearable
-        size="small"
-        type="password"
-        show-password-on="mousedown"
-        :placeholder="t('InputVerifyCode')"
-      />
-    </n-modal>
-
-    <Terminal v-if="shareCode" :share-code="shareCode" />
-  </div>
+  <n-modal
+    v-model:show="showModal"
+    preset="dialog"
+    :title="t('VerifyCode')"
+    :positive-text="t('ConfirmBtn')"
+    :closable="false"
+    :close-on-esc="false"
+    :mask-closable="false"
+    @positive-click="handleConfirm"
+  >
+    <n-input
+      v-model:value="verifyValue"
+      clearable
+      size="small"
+      type="password"
+      show-password-on="mousedown"
+      :placeholder="t('InputVerifyCode')"
+    />
+  </n-modal>
+
+  <TerminalProvider v-if="!showModal">
+    <template #terminal>
+      <Terminal />
+    </template>
+  </TerminalProvider>
 </template>
```

Comment:
There don't seem to be any errors or irregularities present in the provided code snippet, but there may still exist some inefficiencies.

For instance, instead of importing `useConnectionStore` and `nextTick`, you might prefer using `axios.js`. Also, `import Terminal from '@/components/Terminal.vue';` can be improved as it's already imported at layout level in main file so that only once would be required rather than re-importing each times inside components.

Additionally, if necessary, I could suggest an optimization in terms of how verification codes are displayed (for example, changing the placeholder text). Please review this feedback for more accurate information.